py3status/modules/uptime.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+"""
+Display system uptime.
+
+Configuration parameters:
+    days:    display format for days    (default '[\?if=d {d} days, ]')
+    decades: display format for decades (default '[\?if=D {D} decades, ]')
+    format: display format for this module
+        (default 'up {decades}{years}{weeks}{days}{hours}{minutes}')
+    hours:   display format for hours   (default '[\?if=h {h} hours, ]')
+    minutes: display format for minutes (default '[\?if=m {m} minutes, ]')
+    seconds: display format for seconds (default '[\?if=s {s} seconds, ]')
+    weeks:   display format for weeks   (default '[\?if=w {w} weeks, ]')
+    years:   display format for years   (default '[\?if=Y {Y} years, ]')
+
+Format placeholders:
+    {days}    format for days
+    {decades} format for decades
+    {hours}   format for hours
+    {minutes} format for minutes
+    {seconds} format for seconds
+    {weeks}   format for weeks
+    {years}   format for years
+
+Date format placeholders:
+    {D} number of decades
+    {Y} number of years
+    {d} number of days
+    {h} number of hours
+    {m} number of minutes
+    {s} number of seconds
+    {w} number of weeks
+
+Hint: if you don't use one of the placeholders, the value will be carried over
+to the next unit! For example, given an uptime of 1h30min:
+    if you use {minutes} as your only placeholder, then its value will be 90.
+    if you use {hours} and {minutes}, then its values will be 1 and 30, respectively.
+
+@author Alexis "Horgix" Chotard <alexis.horgix.chotard@gmail.com>
+@license BSD
+
+SAMPLE OUTPUT
+{'full_text': 'up 1 days, 18 hours, 20 minutes'}
+"""
+
+from time import time
+STRING_ERROR = "uptime: unknown error"
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    days = '[\?if=d {d} days, ]'
+    decades = '[\?if=D {D} decades, ]'
+    format = 'up {decades}{years}{weeks}{days}{hours}{minutes}'
+    hours = '[\?if=h {h} hours, ]'
+    minutes = '[\?if=m {m} minutes, ]'
+    seconds = '[\?if=s {s} seconds, ]'
+    weeks = '[\?if=w {w} weeks, ]'
+    years = '[\?if=Y {Y} years, ]'
```

Comment:
Let me know if I need to do anything else. I got rid of time formats. We now use one ridiculously awesome long format and does not display unnecessary information.